src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const object = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(object, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        if (key in object) {
+          delete object[key];
+        }
+      }
+    }
+
+    if (action.type === 'clear') {
+      Object.keys(object).forEach((keys) => {
+        delete object[keys];
```

Comment:
In the 'clear' action, using `Object.keys(object).forEach` to delete keys is correct but can be optimized. Consider using `Object.keys(object).forEach(key => delete object[key]);` directly without the intermediate variable `keys` for better readability.